dhctl/pkg/kubernetes/actions/manifests/manifests.go

```diff
@@ -68,190 +74,234 @@ func GetDeckhouseDeployTime(deployment *appsv1.Deployment) time.Time {
 	return deployTime
 }
 
-func ParametrizeDeckhouseDeployment(deployment *appsv1.Deployment, params DeckhouseDeploymentParams) *appsv1.Deployment {
-	deployment.Spec.Template.Spec.Containers[0].Image = params.Registry
+func ParameterizeDeckhouseDeployment(input *appsv1.Deployment, params DeckhouseDeploymentParams) *appsv1.Deployment {
+	deployment := input.DeepCopy()
 
-	deployTime := params.DeployTime
-	if deployTime.IsZero() {
-		deployTime = time.Now()
-	}
+	deckhousePodTemplate := deployment.Spec.Template
+	deckhouseContainer := deployment.Spec.Template.Spec.Containers[0]
+	deckhouseContainerEnv := deckhouseContainer.Env
 
-	for i, env := range deployment.Spec.Template.Spec.Containers[0].Env {
-		switch env.Name {
-		case "LOG_LEVEL":
-			deployment.Spec.Template.Spec.Containers[0].Env[i].Value = params.LogLevel
-		case "DECKHOUSE_BUNDLE":
-			deployment.Spec.Template.Spec.Containers[0].Env[i].Value = params.Bundle
-		case deployTimeEnvVarName:
-			deployment.Spec.Template.Spec.Containers[0].Env[i].Value = deployTime.Format(deployTimeEnvVarFormat)
-		}
-	}
+	freshDeployment := params.DeployTime.IsZero()
 
-	if params.IsSecureRegistry {
-		deployment.Spec.Template.Spec.ImagePullSecrets = []apiv1.LocalObjectReference{
-			{Name: deckhouseRegistrySecretName},
-		}
+	if freshDeployment {
+		params.DeployTime = time.Now()
+	}
 
-		// set volume
+	var (
+		deployTime        bool
+		deployServiceHost bool
+		deployServicePort bool
+	)
+	for _, envEntry := range deckhouseContainerEnv {
+		deployTime = deployTime || envEntry.Name == deployTimeEnvVarName
+		deployServiceHost = deployServiceHost || envEntry.Name == deployServiceHostEnvVarName
+		deployServicePort = deployServicePort || envEntry.Name == deployServicePortEnvVarName
+	}
 
-		volumeToSet := apiv1.Volume{
-			Name: deckhouseRegistryVolumeName,
-			VolumeSource: apiv1.VolumeSource{
-				Secret: &apiv1.SecretVolumeSource{SecretName: deckhouseRegistrySecretName},
+	if !deployTime {
+		deckhouseContainerEnv = append(deckhouseContainerEnv,
+			apiv1.EnvVar{
+				Name:  "KUBERNETES_DEPLOYED",
```

Comment:
I did not see it in last review, but we have const for name https://github.com/deckhouse/deckhouse/pull/716/files#diff-49aea6002f1b4637e82d319b8c945b0eacbaee54c97975123db358619c35cca3R40